### PR TITLE
RUMM-1335 Add basic test cases for Logger API

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In this PR we are adding the first basic nightly tests for the Log public API: `fun v(String, Throwable? = null, Map<String, Any?> = emptyMap())`, `fun d(String, Throwable? = null, Map<String, Any?> = emptyMap())`. The main purpose of this PR was to setup the whole process: Datadog Synthetics Test - > Trigger Bitrise -> Run Nightly Test App -> Monitor each test case output. 
Once this process in place we will start adding the rest of the tests for all the Public APIs.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            rumEnabled = true
+        )
+        return configBuilder.build()
+    }
+
+    // endregion
+
+    companion object {
+        const val ENV_NAME = "instrumentation"
+        const val VARIANT = "nightly"
```
Comment (xgouchet) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt:90]: Variant should be empty. It doesn't make sense to have a value if there's only one variant

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    fun send_a_verbose_log() {
+        val testMethodName = "send_a_verbose_log"
+        val fakeMessage = forge.anAlphaNumericalString()
+        measure(testMethodName) {
+            logger.v(
+                fakeMessage,
+                null,
+                mapOf(
+                    TEST_METHOD_NAME_KEY to testMethodName
+                )
```
Comment (xgouchet) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:50]: You might want to create your attribute outside of the `measure {…}` block. The call to `mapOf()` will include instantiating a map which can skew the measurements.

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        logger = Logger.Builder()
+            .setLoggerName(LOGGER_NAME)
+            .build()
+    }
+
+    /**
+     * apiMethodSignature: Logger#fun v(String, Throwable? = null, Map<String, Any?> = emptyMap())
+     */
+    @Test
+    fun send_a_verbose_log() {
```
Comment (xgouchet) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:41]: This name feels slightly unsound to me (using a natural language syntax, we're creating a log, not sending it. Also we could create a log from a span in the future, shouldn't we disambiguate that). I'd feel more comfortable with a test name that had some kind of link with the API under test. 

Also I'm not sure how the naming will scale, and i'd like each test name to be: 
 - consistent
 - easy to generate for future tests
 - linked with the api under test
 
I didn't see in your RFC a naming convention, but maybe we should define one first to ensure all future tests will be easy to name ?

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        cleanStorageFiles()
+        initializeDatadog()
+        super.start()
+    }
+
+    override fun finish(resultCode: Int, results: Bundle?) {
+        Thread.sleep(DEFAULT_TIMEOUT_IN_MS)
+        Log.i(DatadogTestsRunner::class.java.simpleName, "onTestInstrumentationFinish")
+        Datadog.invokeMethod("stop")
+        cleanStorageFiles()
```
[the remark this answers is not in the document]
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt:40]: just extra safety...maybe I could remove this.

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt (hunk 2 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:50]: Good point

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt (hunk 3 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:41]: Agree, let me add a proposal in the RFC and we can debate there.

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                attributes
+            )
+        }
+    }
+
+    /**
+     * apiMethodSignature: Logger#fun v(String, Throwable? = null, Map<String, Any?> = emptyMap())
+     */
+    @Test
+    fun logs_logger_verbose_log_with_error() {
```
Comment (0xnm) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:58]: shouldn't we also check the case when throwable is there, but no attributes?

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+    // region Runner
+
+    override fun start() {
+        initializeDatadog()
+        super.start()
+    }
+
+    override fun finish(resultCode: Int, results: Bundle?) {
+        Thread.sleep(DEFAULT_TIMEOUT_IN_MS)
```
Comment (0xnm) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt:35]: if I understand the doc of `AndroidJUnitRunner` correctly, `start` and `finish` correspond to the lifecycle of the instrumentation itself and not to the individual tests lifecycle, so this method will be called only once __all__ tests are over and we won't have a delay of 1m after each test?

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+        val testMethodName = "logs_logger_verbose_log_with_error"
```
Comment (0xnm) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:59]: maybe we can also use reflection when `testMethodName` actually matches current test method name? wdyt?

Comment (xgouchet) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:59]: This might be overkill for now, but when the project grows that might be something we can implement eventually

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt (hunk 6 above)
Comment (xgouchet) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt:35]: yes, that's  the goal of it.

**Review by xgouchet — APPROVED** [review]

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt (hunk 7 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:59]: I have a better solution on my end -> AS Live Template
I already created mine I can share it with you also if you want

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt (hunk 5 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsE2ETests.kt:58]: We need attributes no matter what. The `test_method_name` used for monitors it is added in the `special_attributes`

**Review by 0xnm — APPROVED** [review]
